File: Baekjoon/notUploaded/9935.py
str = input();
res = [0]*(len(str)+1);
bomb = input();

idx = 0;
for c in str:
    res[idx] = c;
    if c == bomb[-1]:
        for j in range(len(bomb)-1):
            if bomb[j] != res[idx+j-len(bomb)+1]:
                break;
        else:
            idx = idx-len(bomb);
    idx += 1;
print(''.join(res[0:idx]))

File: Baekjoon/notUploaded/9935.js
const [str,bomb] = require('fs').readFileSync('input.txt').toString().split("\r\n");
//readFileSync('/dev/stdin').toString().split("\n"),
let stack = [];
for (let c of str){
    let top = stack[stack.length-1];
    if (Number.isInteger(top)){
        if (c == bomb.charAt(top)){
            let temp = stack.pop()+1;
            if (temp != bomb.length)
                stack.push(temp);
            continue;
        }
    }
    if (c == bomb.charAt(0)){
        if (bomb.length != 1)
            stack.push(1);
    }else{
        stack.push(c);
    }
}
let res = [];
for (let e of stack){
    if (Number.isInteger(e)){
        for (let i=0; i<e; i++)
            res.push(bomb[i]);
    }else{
        res.push(e);
    }
}
console.log(res.join(''));